### PR TITLE
Add Reversed List Expression

### PR DESCRIPTION
**Pull request by JRoy**
### Description
This pull request adds an expression to reverse a list: `reversed %objects%`

---
**Target Minecraft Versions:** N/A
**Requirements:** N/A
**Related Issues:** #2260

### CODE REVIEW

**Comment by OfficialDonut** [comment]
I believe this was in Skript at one point
https://github.com/SkriptLang/Skript/pull/1406
https://github.com/TheLimeGlass/Skript/blob/bbf7ad02d471fa74fc3da1581fff720149c94054/src/main/java/ch/njol/skript/expressions/ExprReversedList.java

**Comment by JRoy** [comment]
@OfficialDonut That's very strange, maybe Lime was going to pull request it at some point? Anyway, his description is wrong and some of his code is pointless so this PR stands

File: src/main/java/ch/njol/skript/expressions/ExprReversedList.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+		list = (Expression<Object>) exprs[0];
+		return true;
+	}
+	
+	@Nullable
+	@Override
+	protected Object[] get(Event e) {
+		List<Object> reversed = Arrays.asList(list.getAll(e).clone());
+		Collections.reverse(reversed);
```
Comment (bensku) [inline src/main/java/ch/njol/skript/expressions/ExprReversedList.java]: Need to handle errors here. Maybe return nothing if list cannot be reversed?

File: src/main/java/ch/njol/skript/expressions/ExprReversedList.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+
+@Name("Reversed List")
+@Description("Reverses given list.")
```
Comment (bensku) [inline src/main/java/ch/njol/skript/expressions/ExprReversedList.java]: What happens when a list cannot be reversed?

File: src/main/java/ch/njol/skript/expressions/ExprReversedList.java (hunk 1 above)
Comment (JRoy) [reply src/main/java/ch/njol/skript/expressions/ExprReversedList.java]: I had it behave like ExprShuffledList, which will just return the array, unreversed.

File: src/main/java/ch/njol/skript/expressions/ExprReversedList.java (hunk 2 above)
Comment (JRoy) [reply src/main/java/ch/njol/skript/expressions/ExprReversedList.java]: If you're referring to if there is only one element in the list: that would add unneeded complexity to the description. Note that also ExprShuffledList doesn't have this documentation either.

File: src/main/java/ch/njol/skript/expressions/ExprReversedList.java [hunk 3]
```diff
@@ -61,7 +61,11 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 	@Override
 	protected Object[] get(Event e) {
 		List<Object> reversed = Arrays.asList(list.getAll(e).clone());
-		Collections.reverse(reversed);
+		try {
+			Collections.reverse(reversed);
+		} catch (IllegalArgumentException ex) {
+			Skript.error("Tried to reverse a list, but an exception was thrown!");
```
Comment (Blueyescat) [inline src/main/java/ch/njol/skript/expressions/ExprReversedList.java]: Skript.exception....

**Comment by TheLimeGlass** [comment]
I added this before 2.3 to then have the team mess up using git and then having my simple reverse list become the hub for 2.3 and forever got lost and never added for no reason.

**Comment by TheLimeGlass** [comment]
That's what happened with my old pull unfortunately, I think it was when snow was learning git back then.

**Comment by TheLimeGlass** [comment]
I also had the shuffle, which I think Bensku added manually after the whole pull got destroyed.

**Comment by TheLimeGlass** [comment]
Have you tested with variables? When using %object(s)% you need to check that if it's a literal or not, you can use pika's LiteralUtils class to help with this, or use Njol's way of doing it in which he uses the convert to Object which is weird but how he did it.

**Comment by JRoy** [comment]
@bensku @Blueyescat your changes have been applied

**Pull request merged**